### PR TITLE
Update morgue indicator.

### PR DESCRIPTION
**Pull request by eswordthecat**
Pomf told me to import this https://github.com/tgstation/-tg-station/pull/2242, so there you go.

If open (blue)
morgue0

no content (light blue)
morgue1

has a body, but no player (red)
morgue2

items inside, no body (light red)
morgue3

there's a body with a player connected to it (green)
morgue4

Also added the icon_state (existing in the dmi but seems no one used it) of cremator when active
cremator active

### CODE REVIEW

**Comment by Boxe** [comment]
if this does what i think it does then coooool

**Comment by Probe1** [comment]
just what the doctor ordered.

File: code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -21,14 +21,24 @@
...
 
 /obj/structure/morgue/proc/update()
-	if (src.connected)
-		src.icon_state = "morgue0"
+	if (connected)
+		icon_state = "morgue0"
 	else
-		if (src.contents.len)
-			src.icon_state = "morgue2"
+		if (length(contents))
```
Comment (ComicIronic) [inline code/game/objects/structures/morgue.dm]: >length()

File: code/game/objects/structures/morgue.dm [hunk 2]
```diff
@@ -21,14 +21,24 @@
...
+			var/inside = recursive_type_check(src, /mob)
+
+			for (var/mob/body in inside)
+				if (body && body.client)
+					icon_state = "morgue4" // clone that mofo
+					return
+
+			if (length(inside) == 0) // no mobs at all, but objects inside
```
Comment (ComicIronic) [inline code/game/objects/structures/morgue.dm]: >length()

**Comment by ComicIronic** [comment]
Pretty sick however.

File: code/game/objects/structures/morgue.dm (hunk 2 above)
Comment (eswordthecat) [inline code/game/objects/structures/morgue.dm]: Longer than len!

File: code/game/objects/structures/morgue.dm (hunk 1 above)
Comment (eswordthecat) [inline code/game/objects/structures/morgue.dm]: Longer than len!

File: code/game/objects/structures/morgue.dm (hunk 2 above)
Comment (ComicIronic) [inline code/game/objects/structures/morgue.dm]: It's not good code standard!

**Comment by eswordthecat** [comment]
Squashed.

**Pull request merged**